src/Model/Storage/ExternalResource.php

```diff
@@ -45,16 +46,21 @@ public function __construct(IHTTPRequest $httpRequest)
 	/**
 	 * @inheritDoc
 	 */
-	public function get(string $filename)
+	public function get(string $reference)
 	{
-		$parts = parse_url($filename);
+		$data = json_decode($reference);
```

Comment:
What about the pictures that already have been created with a URL string reference? Will they be forever inaccessible because they don’t pass the JSON check?

Reply:
The data isn't stored in any database table. It is only some arrays.

Reply:
Oh, okay.